devolutions-session/src/dvc/window_monitor.rs

```diff
@@ -0,0 +1,635 @@
+//! Window monitoring functionality for tracking active window changes.
+//!
+//! This module provides functionality to monitor the currently focused/active window
+//! on the system, capturing information such as window title, process executable path,
+//! and timestamp (UTC).
+//!
+//! Uses Windows Event Hooks (SetWinEventHook) to receive EVENT_SYSTEM_FOREGROUND
+//! notifications whenever the foreground window changes. Additionally supports optional
+//! polling for detecting title changes within the same window.
+//!
+//! The module provides a channel-based interface for integrating with other systems
+//! (e.g., DVC protocol for transmitting window change events).
+
+use std::cell::RefCell;
+use std::num::NonZeroU64;
+use std::path::PathBuf;
+use std::time::SystemTime;
+
+use anyhow::{Context, Result, bail};
```

Comment:
suggestion: Do not `use anyhow::Result`, instead use `anyhow::Result` everywhere.

rationale: Makes it clear which `Result` it is at the callsite. Don’t shadow the default `std::result::Result` part of the prelude.

This is part of the IronRDP code style.